readthedocs_ext/readthedocs.py

```diff
@@ -172,12 +182,61 @@ def generate_json_artifacts(app, pagename, templatename, context, doctree):
         log.exception(
             'Fail to save JSON output for page {page}'.format(page=outjson)
         )
-    except Exception as e:
+    except Exception:
         log.exception(
             'Failure in JSON search dump for page {page}'.format(page=outjson)
         )
 
 
+def dump_sphinx_data(app, exception):
```

Comment:
I think we should handle the case where `exception` is not `None` and just `return`.